solutions/cses-1641.mdx

```diff
@@ -51,3 +56,77 @@ int main(){
 	cout << "IMPOSSIBLE" << endl;
 }
 ```
+	
+</CPPSection>
+	
+<JavaSection>
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class SumOfThreeValues {
+	static class comparator implements Comparator<Group> {
+		public int compare (Group one, Group two) {
+			return Integer.compare(one.value, two.value);
+		}
+	}
+
+	static class Group {
+		int value;
+		int index;
+	}
+
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int x = io.nextInt();
+		Group[] data = new Group[n];
+
+		for (int i = 1; i <= n; i++) {
+			// store the input in a class that has a value and an index
+			Group group = new Group();
+			group.value = io.nextInt();
+			group.index = i;
+
+			data[i - 1] = group;
+		}
+
+		Arrays.sort(data, new comparator());  // make the values in increasing order
+
+		for (int i = 0; i < n; i++) {
+			int left = 0;
+			int right = n - 1;
+
+			while (left != right) {
+				int target = x - data[i].value;
+				// if the left and the right are not i, their values sum up to the target
+				if (left != i && right != i && data[left].value + data[right].value == target) {
+					// we have an answer, output it
+					io.println(data[i].index + " " + data[left].index + " " + data[right].index);
+
+					io.close();
+					System.exit(0);  // can end the program now
+				}
+				// not equal to target, increase left if the sums are lower
+				if (data[left].value + data[right].value < target) {
+					left++;
+				}
+				// not equal to target, decrease right if the sums
+				// are greater than target
```

Comment:
the comments are line 103, 105, 109, 111, 115, and 116 are unnecessary
what's happening is obvious from looking at the code
you should explain *why* your code is written that way, not just what it's doing

Reply:
the why is already written on the page above

Reply:
spacing is different depending on the person

Reply:
> the why is already written on the page above

1. many people have a hard time matching the why to the implementation, you can even just straight copy paste from the editorial if you're lazy

> spacing is different depending on the person

yeah but yours is just way too spaced apart lmao